src/core/Target.ml

```diff
@@ -12,18 +12,52 @@
  * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the file
  * LICENSE for more details.
  *)
-(*****************************************************************************)
-(* Prelude *)
-(*****************************************************************************)
-(* A target file.
- *
- * See AST_generic.program for more information. This module just
- * defines an alias to have better naming convetion to differentiate
- * a target program from a pattern program.
- *)
 
-(*****************************************************************************)
-(* Types *)
-(*****************************************************************************)
+(* See Target.mli for documentation of public items. *)
+
+type path = { origin : Origin.t; internal_path_to_content : Fpath.t }
+[@@deriving show, eq]
+
+type manifest = { path : path; kind : Manifest_kind.t } [@@deriving show]
+
+type lockfile = {
+  path : path;
+  kind : Lockfile_kind.t;
+  manifest : manifest option;
+}
+[@@deriving show]
+
+type regular = {
+  path : path;
+  analyzer : Xlang.t;
+  products : Semgrep_output_v1_t.product list;
+  lockfile : lockfile option;
+}
+[@@deriving show]
+
+type t = Regular of regular | Lockfile of lockfile [@@deriving show]
+
+let path_of_origin (origin : Origin.t) : path =
+  match origin with
+  | File file -> { origin; internal_path_to_content = file }
```

Comment:
add a TODO saying at some point here we will create temporary files for complex origin like GitBlob

Reply:
Done in d57830ad536d2800a7e4c53159e6c156191eae83.